src/App.jsx

```diff
@@ -14,33 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_LENGTH = 'sortByLength';
+const SORT_ALPHABETICALLY = 'sortAlphabetically';
+const REVERSE_LIST = 'reverseList';
+const RESET = 'reset';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoodsFromServer(goods, buttonId) {
+  switch (buttonId) {
+    case SORT_ALPHABETICALLY:
+      return [...goods].sort((good1, good2) => good1.localeCompare(good2));
+    case SORT_BY_LENGTH:
+      return [...goods].sort((good1, good2) => good1.length - good2.length);
+    case REVERSE_LIST:
+      return [...goodsFromServer].reverse();
+    case RESET:
+      return [...goodsFromServer];
+    default:
+      return goods;
+  }
```

Comment:
The function should be rewrited. Please sort it and after sort make reverse of array in case reverse state is true.